packages/new-deploy/src/DeployContext.ts

```diff
@@ -113,29 +113,97 @@ export default class DeployContext extends MultiProvider<config.Domain> {
-    if (!verification)
-      throw new Error(
-        `Verification with name ${nameOrDomain} for domain ${domain} is not defined`,
-      );
+  // Intended entrypoint.
+  async deployAndRelinquish(): Promise<ethers.PopulatedTransaction[]> {
+    // validate the config input
+    this.validate();
 
-    return verification;
+    // ensure the presence of all core + bridge contracts, and enroll all core + bridge contracts with each other
+    const governanceTransactions = await this.ensureCoreAndBridgeConnections();
+
+    // relinquish control of all other contracts from deployer to governance
+    await this.relinquishOwnership();
+
+    // appoint governor on all networks
+    await Promise.all(
+      this.networks.map((network) =>
+        this.mustGetCore(network).appointGovernor(),
+      ),
+    );
+
+    // perform validation checks on core and bridges
+    await this.checkCores();
+    await this.checkBridges();
+
+    return governanceTransactions;
+  }
+
+  // Checks for connections that are configured but lack a corresponding bridge
+  // connection. Attempts to enroll the bridge routers on eachother.
+  // For any connection that cannot be enrolled, outputs the governance
+  // action required to enroll them
+  protected async ensureCoreAndBridgeConnections(): Promise<
+    ethers.PopulatedTransaction[]
+  > {
+    const governanceTransactions = await this.ensureCoreConnections();
+    const bridgeGovernanceTransactions = await this.ensureBridgeConnections();
+    // combine governance transactions and return them
+    governanceTransactions.push.apply(bridgeGovernanceTransactions);
+    return governanceTransactions;
+  }
+
+  // Deploys all configured connections and enrolls them if possible.
+  // For any connection that cannot be enrolled, outputs the governance
+  // action required to enroll them
+  async ensureCoreConnections(): Promise<ethers.PopulatedTransaction[]> {
+    // ensure all core contracts are deployed
+    await this.ensureCores();
+    // ensure all core contracts are enrolled in each other
+    const enrollTransactions = await Promise.all(
+      this.networks.map(async (network) => {
+        const core = this.mustGetCore(network);
+        const name = this.resolveDomainName(network);
+        const remoteDomains = this.data.protocol.networks[name]?.connections;
+        const txns = await Promise.all(
+          remoteDomains.map((remote) => core.enrollRemote(remote)),
+        );
+        return txns.flat();
+      }),
+    );
+    return enrollTransactions.flat();
+  }
+
+  /// Deploys all configured Cores
+  async ensureCores(): Promise<void> {
+    const networksToDeploy = this.networks.filter((net) => !this.cores[net]);
+    await Promise.all(
+      networksToDeploy.map((net) =>
+        this.deployCore(this.mustGetDomainConfig(net)),
+      ),
+    );
   }
 
   protected async deployCore(domain: config.Domain): Promise<void> {
+    console.log('deployCore ', domain);
+
     this.addDomain(domain);
```

Comment:
is `addDomain` calls `config.addNetwork` - (1) is this fn idempotent? (2) shouldn't the domain already be added if this fn is being called?